deploy/mlflow-triton-plugin/mlflow_triton/deployments.py

```diff
@@ -359,64 +363,68 @@ def _get_copy_paths(self, artifact_path, name, flavor):
         return copy_paths
 
     def _walk(self, path):
-        """Walk a path like os.walk() if path is dir, 
+        """Walk a path like os.walk() if path is dir,
         return file in the expected format otherwise.
         :param path: dir or file path
-        
+
         :return: root, dirs, files
         """
         if os.path.isfile(path):
             return [(os.path.dirname(path), [], [os.path.basename(path)])]
         elif os.path.isdir(path):
             return list(os.walk(path))
         else:
-            raise Exception(f'path: {path} is not a valid path to a file or dir.')
+            raise Exception(
+                f'path: {path} is not a valid path to a file or dir.')
 
     def _copy_files_to_triton_repo(self, artifact_path, name, flavor):
         copy_paths = self._get_copy_paths(artifact_path, name, flavor)
         for key in copy_paths:
-            if 's3' in self.server_config:             
+            if 's3' in self.server_config:
                 # copy model dir to s3 recursively
                 for root, dirs, files in self._walk(copy_paths[key]['from']):
                     for filename in files:
                         local_path = os.path.join(root, filename)
-                        
+
                         if flavor == "onnx":
                             s3_path = os.path.join(
-                                    copy_paths[key]['to'].replace(
-                                    self.server_config['triton_model_repo'], ''), 
-                                    filename,
-                                    ).replace('/', '', 1)
+                                self.server_config['s3_prefix'],
+                                copy_paths[key]['to'].replace(
+                                    self.server_config['triton_model_repo'], '').replace('/', '', 1),
```

Comment:
```suggestion
                                    self.server_config['triton_model_repo'], ''),
```
Seems like the replace here will strip out the slash for version directory, i.e.
`onnx_model_with_files/1/model.onnx -> onnx_model_with_files1/model.onnx`

L0_mflow passed after removing it, @albertoperdomo2 I wonder if you remember why the replace was needed?

Reply:
Thanks for the review and test @GuanLuo 
Just did some experiments to figure out why replace is used previously.

```python
export TRITON_MODEL_REPO=s3://https://oss.vmware.com:443/release/cloud/models
>>> client.create_deployment('onnx_float32_int32_int32', 'models:/onnx_float32_int32_int32/1', flavor="onnx")
print(copy_paths[key]['to'])
s3://release/cloud/models/onnx_float32_int32_int32/1

print(self.server_config['triton_model_repo'])
s3://release/cloud/models

print(copy_paths[key]['to'].replace(self.server_config['triton_model_repo'], ''))
/onnx_float32_int32_int32/1 ==> `replace` is used in this case to trim the first occupation of `/`

print(filename)
model.onnx

print(s3_path)
cloud/models/onnx_float32_int32_int32/1/model.onnx
```

```python
export TRITON_MODEL_REPO=s3://bucket_name
>>> client.create_deployment('onnx_float32_int32_int32', 'models:/onnx_float32_int32_int32/1', flavor="onnx")
print(copy_paths[key]['to'])
s3://bucket_name/onnx_float32_int32_int32/1

print(self.server_config['triton_model_repo'])
s3://bucket_name/

print(copy_paths[key]['to'].replace(self.server_config['triton_model_repo'], ''))
onnx_float32_int32_int32/1

print(filename)
model.onnx

print(s3_path)
onnx_float32_int32_int32/1/model.onnx
```